### PR TITLE
feat(cmd/influx): update password in cli

### LINKED ISSUE #16440: Add the ability to update a user's password from the CLI
as a user
when i use the update capability of the cli
then i would like the ability to update the password of the user

we should add `-p, --password` option to set the password of the user.

```
➜  influxdb git:(master) ✗ ./bin/darwin/influx user update -i 050c7357c0e83000 -h
Update user

Usage:
  influx user update [flags]

Flags:
  -h, --help          Help for the update command
  -i, --id string     The user ID (required)
  -n, --name string   The user name

Global Flags:
      --host string    HTTP address of Influx (default "http://localhost:9999")
      --local          Run commands locally against the filesystem
      --skip-verify    SkipVerify controls whether a client verifies the server's certificate chain and host name.
  -t, --token string   API token to be used throughout client calls
```

### PR DESCRIPTION
**Pull request by kelwang**
Closes #16440

Add the feature, so cli can update user's password. manually tested.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: cmd/influx/setup.go [hunk 1]
```diff
@@ -240,10 +240,14 @@ var (
 	errPasswordIsTooShort = fmt.Errorf("passwords is too short")
 )
 
-func getPassword(ui *input.UI) (password string) {
+func getPassword(ui *input.UI, showNew bool) (password string) {
```
Comment (jademcgough) [inline cmd/influx/setup.go]: If we passed the prompt string instead of using a boolean flag, it'd keep this additional logic out of the function.

Comment (kelwang) [reply cmd/influx/setup.go]: we have 2 promo strings, i think it's over complicated to pass them both. And a little weird to pass only one

File: cmd/influx/user.go [hunk 2]
```diff
@@ -60,28 +103,84 @@ func newUserService() (platform.UserService, error) {
 	}, nil
 }
 
-func userUpdateF(cmd *cobra.Command, args []string) error {
-	s, err := newUserService()
+func newUserSVC() (
+	influxdb.UserService,
+	influxdb.OrganizationService,
+	influxdb.PasswordsService,
```
Comment (jsteenb2) [inline cmd/influx/user.go]: you may want a dep struct for these dependencies, make it easier to maintain

```go
type CmdUserDeps struct {
    userSVC influxdb.UserService
    orgSvc influxdb.OrgService
   ... rest
}
```

File: cmd/influx/user_test.go [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+						Name: "new name",
+					},
+				},
+			},
+			{
+				name: "with password",
+				flags: []string{
+					"--name=new name",
+					"--password=pass1",
+					"--org=org name",
```
Comment (jsteenb2) [inline cmd/influx/user_test.go]: can you add some tests for the env var as part of the `--org` and `--org-id` API contract? Here are some examples from the org service that allow for env vars: https://github.com/influxdata/influxdb/blob/master/cmd/influx/organization_test.go#L247

File: cmd/influx/user_test.go [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+		type called struct {
+			name string
+			id   influxdb.ID
+		}
+
+		tests := []struct {
+			name     string
+			expected called
+			flags    []string
+			envVars  map[string]string
```
Comment (jsteenb2) [inline cmd/influx/user_test.go]: do we need the env vars here? they're never used it looks like

File: cmd/influx/setup.go [hunk 5]
```diff
@@ -200,8 +200,9 @@ var (
 	keyReset    = []byte{keyEscape, '[', '0', 'm'}
 )
 
-func promptWithColor(s string, color []byte) string {
-	return string(color) + s + string(keyReset)
+func promptWithColor(s string, color []byte) []byte {
```
[the remark this answers is not in the document]
Comment (kelwang) [reply cmd/influx/setup.go]: ah, it's the convert of fmt.Println to the ui.Write, which took the bytes slice instead.

**Pull request merged**